### ISSUE TITLE
[needs repro] Unable to switch to nightly-231223

### ISSUE THREAD

**Issue by CodingCellist** [stefan-hoeck/idris2-pack#308]
As part of trying to debug an issue on the discord which was potentially to do with different versions of Idris2 (latest vs release), I tried switching to collection `nightly-231223`, which should be the first collection after [`Idris2-v0.7.0`](https://github.com/idris-lang/Idris2/releases/tag/v0.7.0) was released on the 22nd of December 2023. However, this switch fails:

```
$ pack switch nightly-231223
[ info ] Using package collection nightly-231223
[ cache ] Caching package info for base ............................... ( 1/37)
[ cache ] Caching package info for bounded-doubles .................... ( 2/37)
[ cache ] Caching package info for bounded-doubles-hedgehog-generators  ( 3/37)
<cut>
[ build ] make[1]: Entering directory '/home/thomas/.pack/.tmp0/idris2-compiler/support'
[ build ] make[2]: Entering directory '/home/thomas/.pack/.tmp0/idris2-compiler/support/c'
<cut>
```

Building seems to proceed fine, until it hits:

```
[ build ] 127/285: Building Libraries.Text.PrettyPrint.Prettyprinter.Render.Terminal (src/Libraries/Text/PrettyPrint/Prettyprinter/Render/Terminal.idr)
[ build ] 128/285: Building Libraries.Utils.Term (src/Libraries/Utils/Term.idr)
[ build ] 129/285: Building Idris.Pretty.Render (src/Idris/Pretty/Render.idr)
[ build ] 130/285: Building Idris.Pretty (src/Idris/Pretty.idr)
[ build ] Error: Couldn't parse any alternatives:
[ build ] 1: Unexpected end of expression.
[ build ]
[ build ] Idris.Pretty:311:5--311:17
[ build ]  307 |     prettyPrec d (PDelayed _ _ ty) = parenthesise (d > startPrec) $ "Lazy" <++> prettyPrec appPrec ty
[ build ]  308 |     prettyPrec d (PDelay _ tm) = parenthesise (d > startPrec) $ "Delay" <++> prettyPrec appPrec tm
[ build ]  309 |     prettyPrec d (PForce _ tm) = parenthesise (d > startPrec) $ "Force" <++> prettyPrec appPrec tm
[ build ]  310 |     prettyPrec d (PAutoApp _ f a) =
[ build ]  311 |     parenthesise (d > startPrec) $ group $ prettyPrec leftAppPrec f <++> "@" <+> braces (pretty a)
[ build ]            ^^^^^^^^^^^^
[ build ] ... (2 others)
make: *** [Makefile:69: /home/thomas/.pack/.tmp0/idris2-compiler/build/exec/idris2] Error 1
[ fatal ] Error when executing system command.
          Command: "make" "idris2-exec" "PREFIX=/home/thomas/.pack/install/27780073c8631826d846499840b3857d9b9a4fd5/idris2" "SCHEME=chez"
          Error code: 2
```

* My Chez Scheme executable (v10.0.0) is called `chez` and is on the PATH.
* Idris2 is `$HOME/.pack/bin/idris2`.

I tried `export SCHEME=chez` before running `pack switch`, but unfortunately nothing changed. I'm unsure if the latest version of Idris2 is somehow interfering with bootstrapping the old one (it shouldn't, afaiu), if there is something else I'm missing, or if this is a bug with `pack`. If someone could confirm that this fails on their system, that would be great  : )

**Comment by CodingCellist** [comment]
(moved here from stefan-hoeck/idris2-pack-db#172, apologies once again for the confusion)

**Comment by stefan-hoeck** [comment]
I think this is #303 and has been fixed in #306. Run `pack update` to update the pack executable. Pack should the automatically try bootstrapping the compiler of the nightly in case regular compilation fails. As a faster alternative, run `pack --bootstrap switch nightly-xyz`, which should already work with your version of pack.

**Comment by CodingCellist** [comment]
I can't believe I've been missing `pack update` this whole time :see_no_evil:
I assumed using `latest` would also be autoupdating `pack` since I think of `pack` as a package in and of itself, but I can definitely see why that might be tricky to do (updating your tools while your tools update your tools etc).

#306 works as intended:
```
<same error>
[ warning ] Building Idris failed. Trying to bootstrap now.
<proceeds without issues>
```

I'm including `pack update` as part of my system maintenance now, should avoid future embarrassing mistakes of a similar ilk  : )

Thank you very much for helping with this!
